### PR TITLE
add utility for loading grunt tasks in tests

### PR DESCRIPTION
**Pull request by jacobroufa**
This will be used in mocking certain bits of the grunt application
in order to facilitate testing of the grunt tasks for this repo.

### CODE REVIEW

File: tests/_support/loadGrunt.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import * as grunt from 'grunt';
+import * as path from 'path';
+import * as mockery from 'mockery';
+import * as _ from 'lodash';
```
Comment (devpaul) [inline tests/_support/loadGrunt.ts]: we don't use this

File: tests/_support/loadGrunt.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	[key: string]: any;
+}
+
+export interface TaskLoadingOptions {
+	peerDependencies?: {
+		[key: string]: any;
+	};
+}
+
+export function runGruntTask(taskName: string, callback?: () => void) {
```
Comment (devpaul) [inline tests/_support/loadGrunt.ts]: we talked on ScreenHero about this, but this doesn't apply to our multitasks. We can keep it; it doesn't help us ATM.

File: tests/_support/loadGrunt.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+export function loadTasks(mocks?: MockList, options?: TaskLoadingOptions) {
+	mockery.enable({
+		warnOnReplace: false,
+		warnOnUnregistered: false,
+		useCleanCache: true
+	});
+	mockery.resetCache();
+
+	mockery.registerMock('lodash', _);
```
Comment (devpaul) [inline tests/_support/loadGrunt.ts]: we're not using lodash (although grunt does). I'm guessing `grunt-dojo2` uses and mocks it like this so they can mock a known lodash if/when the need arises.

File: tests/_support/loadGrunt.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+	// Registering this mock as it has problems with `regenerate` from regexpu-core.
+	mockery.registerMock('postcss-modules', function noop() {});
```
Comment (devpaul) [inline tests/_support/loadGrunt.ts]: We don't use this.

File: tests/_support/loadGrunt.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+	if (mocks) {
+		registerMockList(mocks);
+	}
+
+	grunt.registerTask('clean', 'Clean mock task', () => {
+	});
+
+	const packageJson = grunt.file.readJSON('package.json');
+
+	if (options && options.peerDependencies) {
```
Comment (devpaul) [inline tests/_support/loadGrunt.ts]: We don't have peerDependencies

File: tests/_support/loadGrunt.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+	mockery.registerMock('postcss-modules', function noop() {});
+
+	if (mocks) {
+		registerMockList(mocks);
+	}
+
+	grunt.registerTask('clean', 'Clean mock task', () => {
+	});
+
+	const packageJson = grunt.file.readJSON('package.json');
```
Comment (devpaul) [inline tests/_support/loadGrunt.ts]: We don't use our packageJson in our config

File: tests/_support/loadGrunt.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+	if (options && options.peerDependencies) {
+		packageJson.peerDependencies = options.peerDependencies;
+	}
+
+	grunt.file.expand([ '_build/tasks/*.js' ]).forEach((fileName) => {
+		(<any> require).nodeRequire(path.join(process.cwd(), fileName.substr(0, fileName.length - 3)))(grunt, packageJson);
```
Comment (devpaul) [inline tests/_support/loadGrunt.ts]: We don't pass packageJson to our tasks

**Comment by jacobroufa** [comment]
Given the amount of changes required to make this a viable utility for us, and the lack of current necessity, I'm inclined to close this and reinvestigate when it's actually needed at a later time. Thanks for the clarifications @devpaul!

**Pull request closed without merging**